hack/docgen/main.go

```diff
@@ -35,60 +35,68 @@ import (
 )
 
 {{ $tick := "` + "`" + `" -}}
-var (
-	{{ range $struct := .Structs -}}
-	{{ $struct.Name }}Doc encoder.Doc
-	{{ end -}}
-)
 
-func init() {
-	{{ range $struct := .Structs -}}
-	{{ $docVar := printf "%v%v" $struct.Name "Doc" }}
-	{{ $docVar }}.Type = "{{ $struct.Name }}"
-	{{ $docVar }}.Comments[encoder.LineComment] = "{{ $struct.Text.Comment }}"
-	{{ $docVar }}.Description = "{{ $struct.Text.Description }}"
-	{{ range $example := $struct.Text.Examples }}
-	{{ if $example.Value }}
-	{{ $docVar }}.AddExample("{{ $example.Name }}", {{ $example.Value }})
-	{{ end -}}
-	{{ end -}}
-	{{ if $struct.AppearsIn -}}
-	{{ $docVar }}.AppearsIn = []encoder.Appearance{
-	{{ range $value := $struct.AppearsIn -}}
-		{
-			TypeName: "{{ $value.Struct.Name }}",
-			FieldName: "{{ $value.FieldName }}",
+{{ range $struct := .Structs -}}
+func (_ {{ $struct.Name }}) Doc() *encoder.Doc {
```

Comment:
We don't need underscore here. 

```go
func (Config) Doc() *encoder.Doc {
```

works just as fine as

```go
func (_ Config) Doc() *encoder.Doc {
```

Reply:
sure, I will update